packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -4,204 +4,549 @@
-                  ),
-                  Expanded(
-                    child: Container(
-                      margin: const EdgeInsets.only(
-                        left: 16.0,
-                        right: 16.0,
-                        bottom: 16.0,
-                        top: 8.0,
+                    Align(
+                      alignment: Alignment.topRight,
+                      child: Container(
+                        height: topArrowIndicatorHeight,
+                        width: maxWidth - leftArrowIndicatorWidth - margin,
+                        child: ArrowWrapper.unidirectional(
+                          arrowColor: horizontalColor,
+                          child: FittedBox(
+                            child: Text(
+                              properties.horizontalDirectionDescription,
+                              textAlign: TextAlign.center,
+                              textScaleFactor: 1.2,
+                            ),
+                          ),
+                          type: ArrowType.right,
+                        ),
                       ),
-                      child: flexDirectionWrapper,
                     ),
-                  ),
-                ],
+                  ],
+                ),
               );
+            }),
+          ),
+          Container(
+            margin: const EdgeInsets.only(left: 8.0),
+            child: Align(
+              alignment: Alignment.centerLeft,
+              child: _buildAxisAlignmentDropdown(Axis.vertical),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
 
-              final minDimension = min(
-                constraints.maxHeight * 0.5,
-                constraints.maxWidth * 0.5,
-              );
-              final length = min(minDimension, 800.0);
+class WidgetVisualizer extends StatelessWidget {
+  const WidgetVisualizer({
+    Key key,
+    @required this.title,
+    this.hint,
+    @required this.borderColor,
+    this.textColor,
+    this.child,
+  })  : assert(title != null),
+        assert(borderColor != null),
+        super(key: key);
+
+  final String title;
+  final Widget child;
+  final Widget hint;
+
+  final Color borderColor;
+  final Color textColor;
 
-              final flexVisualizerWidget = Container(
-                constraints: BoxConstraints.tight(
-                  Size(
-                    length,
-                    length,
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          IntrinsicHeight(
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+                Container(
+                  constraints: const BoxConstraints(maxWidth: 125.0),
```

Comment:
why is there a maxWidth on this?

Reply:
This is so that the text got constrained and didn't exceed the width for the minimum widget width to be displayed on the screen.
Didn't want to use Flexible/Expanded in this case.
Refactoring this one to reuse the constant

Reply:
great. that makes it self documenting and I don't have to check that the constants are the same.